src/VisualStudio/Core/Def/Implementation/Watson/WatsonReporter.cs

```diff
@@ -57,6 +66,14 @@ public static void Report(string description, Exception exception, Func<IFaultUt
                     arg.AddProcessDump(System.Diagnostics.Process.GetCurrentProcess().Id);
                     return callback(arg);
                 });
+
+            if (exception is OutOfMemoryException)
```

Comment:
this could be changed so that, if exception given to Report is Out of Memory exception, rather than report non fatal watson, it call Fatal.Report so that it explicitly crash VS.